From 037891c1c1cf9f07e438a78c450770609d7af76b Mon Sep 17 00:00:00 2001
From: Hersen Wu <hersenxs.wu@amd.com>
Date: Tue, 12 Jan 2016 12:40:25 -0500
Subject: [PATCH 0688/1110] drm/amd/dal: remove incorrect assert

DRM may remove and add two MST playloads at the same time

Signed-off-by: Hersen Wu <hersenxs.wu@amd.com>
Acked-by: Jordan Lazare <Jordan.Lazare@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index 8bd254c..a994a4c 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -1360,8 +1360,6 @@ static enum dc_status allocate_mst_payload(struct core_stream *stream)
 	}
 
 	ASSERT(proposed_table.stream_count > 0);
-	ASSERT(proposed_table.stream_count -
-			link->stream_alloc_table.stream_count == 1);
 
 	/*
 	 * temporary fix. Unplug of MST chain happened (two displays),
@@ -1457,9 +1455,6 @@ static enum dc_status deallocate_mst_payload(struct core_stream *stream)
 		proposed_table.stream_allocations[i].slot_count);
 	}
 
-	ASSERT(link->stream_alloc_table.stream_count -
-			proposed_table.stream_count == 1);
-
 	dc->hwss.update_mst_stream_allocation_table(
 		link_encoder,
 		&proposed_table);
-- 
2.7.4

